orc8r/cloud/go/obsidian/swagger/specs.go

```diff
@@ -0,0 +1,88 @@
+/*
+ Copyright 2021 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+// readEntireFileString wraps ioutil.ReadFile and returns a string iff the
+// entire file can be read without error.
+func readEntireFileString(path string) (string, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+type specType string
+
+const (
+	partial    specType = "partial"
+	standalone specType = "standalone"
+)
+
+func (t specType) path(base string, service string) string {
+	return filepath.Join(
+		base,
+		string(t),
+		fmt.Sprintf("%s.swagger.v1.yml", strings.ToLower(service)))
+}
+
+// Specs provide access to swagger spec YAML files.
+type Specs interface {
+	GetCommonSpec() (string, error)
+	GetPartialSpec(service string) (string, error)
+	GetStandaloneSpec(service string) (string, error)
+}
+
+// fsSpecs implements Specs via the filesystem at a specific base dir.
+type fsSpecs struct {
+	baseDir string
+}
+
+var _ Specs = (*fsSpecs)(nil)
```

Comment:
Don't think this is necessary, since we have `NewSpecs` below

Reply:
I go back and forth. So far I've preferred to do the type assertion in the `New*` functions and call it a day. I think for uniformity let's just go with that, but I'm generally open to being convinced